### DISCUSSION TITLE
Quarkus and Unit testing

### DISCUSSION THREAD

**Discussion by konradgab** [quarkusio/quarkus#25543] [Q&A] [answered]
Hi 
I would like to ask if there is any recommended annotation for writing unit tests or it is necesary to add dependencies to junit (despite having quarkus-junit5 dependencies)
I read about @QuarkusUnitTest at documentation but can't use it.
My case is that i have some properties to for example keycloak so if i use @QuarkusTest to run unit test I will depend on the correct work of the keycloak (because the test will run on a quarkus instance) which is undesirable.

**Comment by maxandersen** [comment, marked as answer]
@QuarkusUnitTest is for internal quarkus testing usage afaik.
 
 Why do you find you cannot use something like @QuarkusTest (which yes is integrated with junit) ?
